fenix/app/src/main/java/org/mozilla/fenix/tabstray/browser/compose/ReorderableGrid.kt

```diff
@@ -0,0 +1,314 @@
+        }
+        val startOffset = draggingItem.offset.toOffset() + draggingItemOffset
+        val endOffset = Offset(
+            startOffset.x + draggingItem.size.toSize().width,
+            startOffset.y + draggingItem.size.toSize().height,
+        )
+        val middleOffset = startOffset + (endOffset - startOffset) / 2f
+
+        val targetItem = gridState.layoutInfo.visibleItemsInfo.find { item ->
+            middleOffset.x.toInt() in item.offset.x..item.endOffset.x &&
+                middleOffset.y.toInt() in item.offset.y..item.endOffset.y &&
+                draggingItemKey != item.key
+        }
+        if (targetItem != null && targetItem.key !in ignoredItems) {
+            if (draggingItem.index == gridState.firstVisibleItemIndex) {
+                scope.launch {
+                    gridState.scrollBy(-draggingItem.size.height.toFloat())
+                }
+            }
+            onMove.invoke(draggingItem, targetItem)
+        } else {
+            val overscroll = when {
+                draggingItemCumulatedOffset.y > 0 ->
+                    (endOffset.y - gridState.layoutInfo.viewportEndOffset).coerceAtLeast(0f)
+
+                draggingItemCumulatedOffset.y < 0 ->
+                    (startOffset.y - gridState.layoutInfo.viewportStartOffset).coerceAtMost(0f)
+
+                else -> 0f
+            }
+            if (overscroll != 0f) {
+                scope.launch {
+                    gridState.scrollBy(overscroll)
+                }
+            }
+        }
+    }
+}
+
+/**
+ * Container for draggable grid item
+ *
+ * @param state State of the lazy grid
+ * @param key Key of the item to be displayed
+ * @param content Content of the item to be displayed
+ */
+@ExperimentalFoundationApi
+@Composable
+fun LazyGridItemScope.DragItemContainer(
+    state: GridReorderState,
+    key: Any,
+    content: @Composable () -> Unit,
+) {
+    val modifier = when (key) {
+        state.draggingItemKey -> {
+            Modifier
+                .zIndex(1f)
+                .graphicsLayer {
+                    translationX = state.draggingItemOffset.x
+                    translationY = state.draggingItemOffset.y
+                }
+        }
+
+        state.previousKeyOfDraggedItem -> {
+            Modifier
+                .zIndex(1f)
+                .graphicsLayer {
+                    translationX = state.previousItemOffset.value.x
+                    translationY = state.previousItemOffset.value.y
+                }
+        }
+
+        else -> {
+            Modifier
+                .zIndex(0f)
+                .animateItemPlacement(tween())
+        }
+    }
+    val interactingModifier = if (state.interactingItemKey == key) {
+        Modifier.indication(
+            state.interactionSource,
+            rememberRipple(
+                color = when (isSystemInDarkTheme()) {
+                    true -> PhotonColors.White
+                    false -> PhotonColors.Black
+                },
```

Comment:
I remember when we first introduced them we [intended](https://github.com/mozilla-mobile/firefox-android/pull/2578#discussion_r1238738141) to keep them manual since they're not provided by the design system. Should we change that now?

Reply:
gotcha gotcha, I'm cool with deferring this for now, since we just had that convo you linked above not too long ago, and we haven't had that conversation with UX yet.